Preserve null in babel-types' clone and deepClone.

NTillmann opened the pull request:
| Q                 | A <!--(yes/no) -->
| ----------------- | ---
| Bug fix?          | Yes
| Breaking change?  | No
| New feature?      | No
| Deprecations?     | No
| Spec compliancy?  | No
| Tests added/pass? | Yes
| Fixed tickets     | Fixes #4852
| License           | MIT
| Doc PR            | 
| Dependency Changes| 

Preserve null in babel-types' clone and deepClone.
(The current behavior turns null into {}, which is not equivalent. Null is a valid expression used to represent missing array elements. The Printer in particular cannot generate code for {}, but it does handle null.)
Add tests.

hzoo commented:
Cool - what about a test case for a node that contains null for deepClone and I assume  undefined is good?

hzoo commented:
I would add your original test case in the tests? `[,0]` (I removed the quotes in the description)

NTillmann commented:
Thanks for the feedback! I updated the implementation to also handle undefined, and added tests for top-level undefined and nested explicit null. However, I don't think there is any valid nested place for undefined --- if you know one, I could add a test for it. Otherwise, as far as I know, null is really the only special value where things went wrong before.

hzoo commented:
Awesome! Thanks for contributing 👌
